libs/azure-api-client/src/ApiClient.php

```diff
@@ -28,13 +31,14 @@ class ApiClient
 
     private readonly HandlerStack $requestHandlerStack;
     private readonly GuzzleClient $httpClient;
+    private readonly AuthenticatorInterface $authenticator;
     private readonly LoggerInterface $logger;
 
     /**
      * @param array{
      *     baseUrl?: null|non-empty-string,
      *     backoffMaxTries?: null|int<0, max>,
-     *     middleware?: null|list<callable>,
+     *     authenticator?: null|AuthenticatorInterface,
      *     requestHandler?: null|callable,
      *     logger?: null|LoggerInterface,
```

Comment:
nemohl by byt authenticator a logger prostě named param? Celkově by měli ty options array srance umřít.

Reply:
taky bych se toho nebál, s PHP8 to může mít argumentů kolik chce

Reply:
ok, dame named argument :)